File: PTT-GC-API/Dtos/Initiative/InitiativeList.cs
using System;
using System.Collections.Generic;
using PTT_GC_API.Models.Initiative;

namespace PTT_GC_API.Dtos.Initiative
{
    public class InitiativeList
    {
        public int Id { get; set; }
        public string InitiativeCode { get; set; }
        public string Name { get; set; }
        public string OwnerName { get; set; }
        public string Organization { get; set; }
        public DateTime? RegisteringDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public bool? Cim { get; set; }
        public bool? Pim { get; set; }
        public bool? Dim { get; set; }
        public bool? Max { get; set; }
        public bool? DirectCapex { get; set; }
        public bool? Cpi { get; set; }
        public bool? Strategy { get; set; }
        public bool? RandD { get; set; }
        public bool? Other { get; set; }
        public string InitiativeType { get; set; }
        public string Status { get; set; }
        public string Stage { get; set; }
        public bool? IsAddMore { get; set; }
        public bool? IsRevise { get; set; }
        public bool? IsReturn { get; set; }
        public string WbsNo { get; set; }
        public string Company { get; set; }
        public string AssignTo { get; set; }
        public int Counter { get; set; }
        public string SapStatus { get; set; }
        public string MocStatus { get; set; }
        public virtual ICollection<InitiativeAction> InitiativeActions { get; set; }
    }
}

File: PTT-GC-SPA/src/app/core/models/initiativeList.ts
export interface InitiativeList {
  id: number;
  initiativeCode: string;
  name: string;
  ownerName: string;
  organization: string;
  registeringDate: Date;
  cim: number;
  pim: number;
  dim: number;
  max: number;
  directCapex: number;
  cpi: number;
  strategy: number;
  randD: number;
  other: number;
  initiativeType: string;
  status: string;
  stage: string;
  createdBy: string;
  createdDate: Date;
  isAddMore: boolean;
  isRevise: boolean;
  isReturn: boolean;
  updatedDate: Date;
  wbsNo: string;
  company: string;
  assignTo: string;
  counter: number;
}

/**
 * Initiative ID
Initiative Name
Owner Name
Investment type​
Benefit Type​ (IRR/Jfactor)
Benefit Value​
Project Cost (M THB)
Stage Gate
Reason (Concern Issue/Necessity)​
 */

export interface InitiativeListPoolPim {
  selected: boolean;
  id: number;
  initiativeCode: string;
  name: string;
  ownerName: string;
  typeOfInvestment: string;
  typeBenefit: string;
  benefit: number;
  costEstimate: number;
  stage: string;
  note: string;
}

export interface SearchConditonPoolPim {
  organization: string;
  plant: string;
  gate: string;
}

export interface InitiativeMemberVACPIC {
  vacPicId: number;
  memberType: string; // pic, vac
  initiativeId: number;
  initiativeCode: string;
  name: string;
  ownerName: string;
  initiativeType: string;
  plant: string;
  emocNo: string;
  gate: string;
  presentation: string;
  pdd: string;
  overallCostEst: number;
  requestCapex: number;
  costEstCapexType: number;
  requestOpex: number;
  fxExchange: number;
  costEstOpexType: string;
  result: string;
  stage: string;
  process: string;
  tnitiativeStatus: string;
}